scripts/deploy.sh

```diff
@@ -209,7 +229,33 @@ function deploy_app_wallet() {
 
   echo "Deploying the Orbit Wallet to the '$network' network."
 
-  BUILD_MODE=$network dfx deploy --network $network app_wallet --with-cycles 2000000000000 $([[ -n "$subnet_type" ]] && echo "--subnet-type $subnet_type")
+  if should_build_artifacts || [ ! -f ./artifacts/wallet-dapp/wallet-dapp.tar.gz ]; then
+    ./scripts/docker-build.sh --wallet-dapp
+  fi
+
+  if [ -d ./artifacts/wallet-dapp/dist ]; then
+    rm -rf ./artifacts/wallet-dapp/dist
+  fi
+
+  mkdir -p ./artifacts/wallet-dapp/dist
+  tar -xvf ./artifacts/wallet-dapp/wallet-dapp.tar.gz -C ./artifacts/wallet-dapp/dist
+
+  set +e # Disable 'exit on error'
+  canister_id_output=$(dfx canister id app_wallet --network $network 2>&1)
+  canister_id_exit_code=$?
+  set -e # Re-enable 'exit on error'
+
+  if [ $canister_id_exit_code -ne 0 ]; then
+    echo "Canister 'app_wallet' does not exist, creating and installing..."
+
+    BUILD_MODE=$network dfx deploy --network $network app_wallet --with-cycles 2000000000000 $([[ -n "$subnet_type" ]] && echo "--subnet-type $subnet_type")
+  else
+    echo "Deploying the app_wallet canister to the '$network' network is not yet supported automatically."
+    echo
+    echo "Please deploy the app_wallet canister manually using the following command:"
+    echo "icx-asset --pem YOUR_IDENTITY_PEM_PATH --replica TARGET_REPLICA_HTTP sync --no-delete $canister_id_output artifacts/wallet-dapp/dist"
```

Comment:
Can't we just run this here instead of failing? If we do want to fail, maybe `exit 1` would be appropriate here?

Reply:
Actually I think we could require the identity pem path as an argument and that would enable us to run the command instead of just logging it, I’ll try that.